src/V3Timing.cpp

```diff
@@ -153,13 +165,10 @@ class TimingSuspendableVisitor final : public VNVisitor {
             if (!cextp->classp()->user1SetOnce()) cextp->classp()->repairCache();
             if (auto* const overriddenp
                 = VN_CAST(cextp->classp()->findMember(nodep->name()), CFunc)) {
-                if (overriddenp->user2()) {  // If it's suspendable
-                    nodep->user2(true);  // Then we are also suspendable
-                    // As both are suspendable already, there is no need to add it as our
-                    // dependency or self to its dependencies
-                } else {
-                    // Make a dependency cycle, as being suspendable should propagate both up
-                    // and down the inheritance tree
+                if (nodep->user2() < overriddenp->user2()) {
+                    nodep->user2(overriddenp->user2());  // Then we are also suspendable
+                }
+                if (nodep->user2() < PROC) {  // Add a vertex only if the flag can still change
```

Comment:
I think the comment about the dependency cycle is still useful here.

Reply:
Right. I'be brought it back.